graylog2-web-interface/src/views/components/datatable/bindings.tsx

```diff
@@ -14,13 +14,47 @@
  * along with this program. If not, see
  * <http://www.mongodb.com/licensing/server-side-public-license>.
  */
+
 import type { VisualizationType } from 'views/types';
 import DataTable from 'views/components/datatable/DataTable';
+import DataTableVisualizationConfig from 'views/logic/aggregationbuilder/visualizations/DataTableVisualizationConfig';
+import type { WidgetConfigFormValues } from 'views/components/aggregationwizard';
 
+type DataTableVisualizationConfigFormValues = {
+  pinned_columns: Array<string>,
+};
 const dataTable: VisualizationType<typeof DataTable.type> = {
   type: DataTable.type,
   displayName: 'Data Table',
   component: DataTable,
+  config: {
+    createConfig: () => ({ pinned_columns: [] }),
+    fromConfig: (config: DataTableVisualizationConfig | undefined) => ({ pinned_columns: config?.pinned_columns.toJS() ?? [] }),
+    toConfig: (formValues: DataTableVisualizationConfigFormValues) => DataTableVisualizationConfig.create(formValues.pinned_columns),
+    fields: [{
+      name: 'pinned_columns',
+      title: 'Pinned Columns',
+      type: 'multi-select',
+      options: ({ formValues }: { formValues: WidgetConfigFormValues }) => {
+        const options = formValues?.groupBy?.groupings.reduce((res, cur) => {
```

Comment:
We can make this a bit simpler like this:
```suggestion
        const options = formValues?.groupBy?.groupings.filter((grouping) => (grouping?.direction === 'row' && grouping?.field?.field))
          .map(grouping => grouping.field.field) ?? [];
```

Reply:
With `reduce`, we iterate  through the array only once

Reply:
We can expect that the number of groupings is small enough so that the memory impact is negligible, especially because it is a shallow copy. For this case, simplicity of the code trumps a small memory saving.